slang-api/src/test/java/com/sonarsource/slang/impl/UnaryExpressionTreeImplTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.impl;
+
+import com.sonarsource.slang.api.NativeKind;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.api.TreeMetaData;
+import com.sonarsource.slang.api.UnaryExpressionTree;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static com.sonarsource.slang.utils.SyntacticEquivalence.areEquivalent;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UnaryExpressionTreeImplTest  {
+
+  private class TypeNativeKind implements NativeKind {}
+
+  @Test
+  public void test() {
+    TreeMetaData meta = null;
+    Tree condition = new IdentifierTreeImpl(meta, "x");
+    Tree negCondition = new UnaryExpressionTreeImpl(meta, UnaryExpressionTree.Operator.NEGATE, condition);
+    Tree nativeTree = new NativeTreeImpl(meta, new TypeNativeKind(), Arrays.asList(condition));
+    Tree negNative = new UnaryExpressionTreeImpl(meta, UnaryExpressionTree.Operator.NEGATE, nativeTree);
+
+    assertThat(negCondition.children()).containsExactly(condition);
+    assertThat(areEquivalent(negCondition, negCondition)).isTrue();
```

Comment:
You can use `assertTree(x).isEquivalentTo(y)` instead of testing the resulting boolean.
Also, I would prefer to have another instance of `negCondition` for testing equivalence, as this case will only test the `first == second` in the `areEquivalent` method

Reply:
We cannot use assertTree here because it would create a cycle in the dependencies.